### PR TITLE
Create Feed Content index for Articles and Podcast Episodes and User Index

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Currently, in Algolia, Articles, Podcast Episodes, and Users are all in the same "searchables" index. Podcast Eps and Articles actually share a lot of the same fields when we index them so having them on the same index makes sense. However, users do not. This creates an index for Articles and Podcast Eps and a separate index for Users. 

Eventually, all of these should have their own separate dedicated search areas, but for now we can issue a search to multiple indexes from the Feed page or we can issue 2 searches if no class_name filter is present. I will be playing around to see what makes the most sense. 

I also took the time to make a base_spec class to handle testing the base search class functionality making the additional two search class specs easier to write. Eventually we should refactor the rest of the search model specs. 

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-32289326
https://github.com/thepracticaldev/dev.to/projects/6#card-33956385

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

alt_text

### CODE REVIEW

File: app/services/search/feed_content.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    INDEX_ALIAS = "feed_content_#{Rails.env}_alias".freeze
+    MAPPINGS = JSON.parse(File.read("config/elasticsearch/mappings/feed_content.json"), symbolize_names: true).freeze
+
+    class << self
+      private
+
+      def index_settings
+        if Rails.env.production?
+          {
+            number_of_shards: 10,
```
Comment (atsmith813) [inline app/services/search/feed_content.rb]: There's not a solid "rule of thumb" for something like this, right? It's arbitrary?

File: config/elasticsearch/mappings/feed_content.json [hunk 2]
```diff
@@ -0,0 +1,165 @@
+{
```
Comment (atsmith813) [inline config/elasticsearch/mappings/feed_content.json]: I love that we have this unspoken/understood rule going of ordering the mappings with `id` and then the rest of the properties in alphabetical order 💯

File: spec/services/search/base_spec.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
```
Comment (atsmith813) [inline spec/services/search/base_spec.rb]: +1 for this whole concept!

File: app/services/search/feed_content.rb (hunk 1 above)
Comment (mstruve) [reply app/services/search/feed_content.rb]: Yeah, it's based on how many I think will last us for a while but also not use a ton of resources right off the bat

File: spec/services/search/base_spec.rb (hunk 3 above)
Comment (mstruve) [reply spec/services/search/base_spec.rb]: Figured it was about time 😉

**Comment by mstruve** [comment]
lol I definitely didnt want searchables since everything is searchable. I considered published content but classified listings are also technically published. I think once our search evolves grouping and name these will likely become more clear.

**Pull request merged**